debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordState.java

```diff
@@ -365,13 +378,19 @@ else if (simpleFieldName.equals(TransactionMonitor.DEBEZIUM_TRANSACTION_ID_KEY)
             else {
                 return Arrays.stream(addHeadersConfig.split(","))
                         .map(String::trim)
-                        .map(field -> new FieldReference(fieldPrefix, field))
+                        .map(field -> {
+                            String[] split = field.split(":");
+                            if (split.length > 1) {
+                                return new FieldReference(fieldPrefix, split[0].trim(), split[1].trim());
+                            }
+                            return new FieldReference(fieldPrefix, field);
+                        })
                         .collect(Collectors.toList());
             }
         }
 
-        String getNewFieldName() {
```

Comment:
How about returning an object here which encapsulates the renaming logic. I think `FieldReference` already is that type pretty much, just push the logic from `map()` above to there.

Reply:
I have refactored and now I don't think i can return object instead. 
Now the code looks quite concise.